slo-monitor/src/monitors/pod_monitor.go

```diff
@@ -133,6 +139,15 @@ func (pm *PodStartupLatencyDataMonitor) Run(stopCh chan struct{}) {
 			pm.handlePodUpdate(p)
 			return nil
 		},
+		func(obj interface{}) error {
+			p, ok := obj.(*v1.Pod)
+			if !ok {
+				glog.Errorf("Failed to cast observed object to *v1.Pod: %v", obj)
+				return nil
+			}
```

Comment:
This comment should have been below :)

But anyway - this handles only getting the name. But since you're trying to get the  pod, this doesn't handle it.